osu.Game/Online/Chat/MessageFormatter.cs

```diff
@@ -246,6 +246,44 @@ public static LinkDetails GetLinkDetails(string url)
             return new LinkDetails(LinkAction.External, url);
         }
 
+        public static string? GetUrl(LinkDetails link)
+        {
+            switch (link.Action)
+            {
+                case LinkAction.External:
+                    return (string)link.Argument;
```

Comment:
Should handle osu! links without root url:
![osu!_BE6muXsviO](https://github.com/ppy/osu/assets/35318437/ba03ce0b-054e-4998-9f85-a561be54b015)